src/SideNavigation/SideNav.test.js

```diff
@@ -374,4 +374,23 @@ describe('<SideNav />', () => {
             ).toBe('Sample');
         });
     });
+
+    describe('onItemSelect handler', () => {
+        test('should dispatch the onItemSelect callback with the event', () => {
+            let f = jest.fn();
+
+            const element = mount(<SideNav data-sample='Sample' onItemSelect={f}>
+                <SideNav.List>
+                    <SideNav.ListItem
+                        id='item-1'
+                        name='Link Item 1'
+                        url='#' />
+                </SideNav.List>
+            </SideNav>);
+
+            element.find('#item-1 a').simulate('click');
+
+            expect(f).toHaveBeenCalledTimes(1);
```

Comment:
Although this test is good, I think it might be worth also adding a check to make sure the second argument is the proper `id` value (you can use `toHaveBeenCalledWith`).